### DISCUSSION TITLE
Community plugins for Enterprise edition

### DISCUSSION THREAD

**Discussion by zmstone** [emqx/emqx#4380] [Ideas]
This is a discussion thread to follow up on https://github.com/emqx/emqx/issues/3998#issuecomment-802781070

Community plugins are supported in `lib-extra` (see more details in the [README](https://github.com/emqx/emqx/blob/master/lib-extra/README.md))

However we do not have a plan to bring them into Enterprise edition due to limited resource.

Our long term goal is to create a market-place in EMQ X dashboard where you can click (or call API)
to download / install / load plugins. --- for both open source and enterprise.
Before that, we need to think of a solution how we can bring community plugins to EMQ X Enterprise

Ideas are welcome.

**Comment by zmstone** [comment]
One not so ideal solution is to support building plugin tarballs from opensource project.
Then with an install script, we can extract it to the install paths of EMQ X (opensource or enterprise).

**Comment by endianjoakim** [comment]
I assume Enterprise customers would be interested in paying to have some additional plugins available. Perhaps offer community plugins evaluation and compilation as a service to Enterprise customers.

**Reply by zmstone** [reply]
Yes sure. For the plugins with a compliant open source license, sufficient tests, well documented, we'd be happy to have them included in enterprise edition.

The release cycle is however much longer, and we have limited  capacity etc.
Have a more flexible way to allow user's to add plugins by their own seem to be a well balance.?

**Comment by zmstone** [comment]
EMQX and EMQX Enterprise v5 supports building plugin package independently and install the package from dashboard.
see more: https://www.emqx.io/docs/en/latest/extensions/plugins.html#develop-emqx-plugins
